index.bs

```diff
@@ -273,73 +273,68 @@ requestFontsButton.onclick = async function() {
+</aside>
 
 
 <!-- ============================================================ -->
-## Name Table ## {#concept-name-table}
+# Permissions Integration # {#permissions-integration}
 <!-- ============================================================ -->
 
-A [=/font representation=] has a <dfn for="font representation">name table</dfn>, which is the [=/font table=] in its [=font representation/table list=] with [=font table/tag=] \``name`\`.
-
-The [=font representation/name table=] has a [=/map=] <dfn for="name table">names</dfn>, which is a mapping from an {{unsigned short}} (a <dfn>name ID</dfn>) to a [=/localized string map=]. [=/Name IDs=] are defined as part of the [[!OPENTYPE]] <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/name">naming table</a> specification.
-
-A <dfn>localized string map</dfn> is a [=/map=] from a BCP 47 language tag to a {{USVString}}. [[BCP47]]
-
-Note: [[!OPENTYPE]] does not actually specify the usage of [[BCP47]] language tags; instead, platform specific language IDs are used, e.g. US English is language ID = 0x409 for Windows and language ID = 0 for Macintosh. This specification assumes a mapping exists.
+Enumeration of local fonts requires a permission to be granted.
 
-<div algorithm >
+<!-- ============================================================ -->
+## System Fonts ## {#concept-system-fonts}
+<!-- ============================================================ -->
 
-A [=/font representation=]'s <dfn for="font representation" lt="name string">name string |id| for |language|</dfn> is given by these steps:
+A <dfn>system font</dfn> is a font that is available system-wide provided by the operating system.
 
-1. Let |names| be the [=/font representation=]'s [=font representation/name table=]'s [=name table/names=].
-1. Let |name| be |names|[|id|], a [=/localized string map=].
-1. If |name|[|language|] [=map/exists=], then return it.
-1. Otherwise, return the first entry in |name|.
+To <dfn lt="read as font representation">read a system font as a font representation</dfn> is to provide a [=/font representation=] equivalent to the font, or failure otherwise. This means providing all properties of a [=/font representation=]:
+* [=font representation/Data bytes=].
+* [=font representation/PostScript name=], which must be valid.
+* [=font representation/Full name=].
+* [=font representation/Family name=].
+* [=font representation/Style name=].
 
-Issue: What if there is no matching |id|? Where does fallback to a different name occur?
 
-Issue: Matching scheme needs to be defined, with the caveat that some OSes don't expose the data here.
+<div algorithm>
+To <dfn>get all system font representations</dfn>, run these steps:
+1. Let |fonts| be a subset of [=/system fonts=].
+1. Let |result| be a new [=/list=].
+1. [=list/For each=] |font| in |fonts|.
+    1. Let |representation| be |font| [=/read as font representations=]. On failure, [=iteration/continue=].
```

Comment:
"Let representation be font read as font representations." doesn't flow well ... do you mean "read as a font representation"?